src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1312,16 +1312,20 @@ private function dumpValue($value, $interpolate = true)
                 }
 
                 if (is_array($factory)) {
+                    $factoryMethod = substr($this->dumpValue($factory[1]), 1, -1);
+
                     if (is_string($factory[0])) {
-                        return sprintf('\\%s::%s(%s)', $factory[0], $factory[1], implode(', ', $arguments));
+                        $factoryClass = substr(str_replace('\\\\', '\\', $this->dumpValue($factory[0])), 1, -1);
```

Comment:
Are you sure? I was wondering why my config was failing when migrating to the new syntax. And something similar is already done when using the old syntax: https://github.com/symfony/symfony/blob/2.6/src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php#L1331-1339